lib/query-conversation/src/main/java/io/redlink/smarti/query/conversation/ConversationSearchService.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 Redlink GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.redlink.smarti.query.conversation;
+
+import io.redlink.smarti.api.StoreService;
+import io.redlink.smarti.model.Client;
+import io.redlink.smarti.model.Conversation;
+import io.redlink.smarti.model.SearchResult;
+import io.redlink.solrlib.SolrCoreContainer;
+import io.redlink.solrlib.SolrCoreDescriptor;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static io.redlink.smarti.query.conversation.ConversationIndexConfiguration.FIELD_OWNER;
+import static io.redlink.smarti.query.conversation.ConversationIndexConfiguration.FIELD_TYPE;
+
+@Service
+public class ConversationSearchService {
+
+    private final SolrCoreContainer solrServer;
+    private final SolrCoreDescriptor conversationCore;
+
+    private final StoreService storeService;
+
+    @Autowired
+    public ConversationSearchService(SolrCoreContainer solrServer, @Qualifier(ConversationIndexConfiguration.CONVERSATION_INDEX) SolrCoreDescriptor conversationCore, StoreService storeService) {
+        this.solrServer = solrServer;
+        this.conversationCore = conversationCore;
+        this.storeService = storeService;
+    }
+
+
+    public SearchResult<Conversation> search(Client client, MultiValueMap<String, String> queryParams) throws IOException {
+
+        final ModifiableSolrParams solrParams = new ModifiableSolrParams(toListOfStringArrays(queryParams, "text"));
```

Comment:
There is not much semantic within the queryParams. I could imagine that at this point it would be great to have some more meaningful "ConversationQuery" object. Or in other words: If I should make a suggestion at wich point of code I would expect most changes in future, than I would point on this part, where the Solr query is being constructed.

My question: Should we spend some efforts on how to create the query? Or should we create an improvement issue for this? Or do you think, that just using `HTTP query Strings` and `ModifiableSolrParams` is sufficient?

Reply:
I think we currently have good trade-off between simplicity and flexibility:
* the `text` parameter allows simple full-text search on conversations
* advanced queries can be executed by using the Solr-parameters.

If you identify additional features that should be exposed through explicit parameters (e.g. date-filters, participants, etc - just guessing) on the web-service level we can add those.

On the service level, I'd stay with the current api for now - we can add a method accepting a `ConvesationSearchQuery` (and a matching `ConversationSearchQueryBuilder`) when we need it.

Reply:
👍